tests/v2/validation/workloads/workload_test.go

```diff
@@ -192,6 +200,159 @@ func (w *WorkloadTestSuite) TestWorkloadStatefulset() {
 	require.NoError(w.T(), err)
 }
 
+func (w *WorkloadTestSuite) TestWorkloadUpgrade() {
+	subSession := w.session.NewSession()
+	defer subSession.Cleanup()
+
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(w.client, w.cluster.ID)
+	require.NoError(w.T(), err)
+
+	nginxDeployment, err := deployment.CreateDeployment(w.client, w.cluster.ID, namespace.Name, 2, "", "", false, false)
+	require.NoError(w.T(), err)
+
+	err = charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespace.Name, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + nginxDeployment.Name,
+	})
+	require.NoError(w.T(), err)
+
+	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespace.Name, nginxDeployment)
+	require.NoError(w.T(), err)
+
+	revisions, err := deployment.GetRolloutHistoryDeployment(w.client, w.cluster.ID, namespace.Name, nginxDeployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), []string{"1"}, revisions)
+
+	countPods, err := pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, "nginx")
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+
+	containerName := namegen.AppendRandomString("updatetestcontainer")
+	newContainerTemplate := workloads.NewContainer(containerName,
+		"redis",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	redisDeployment := nginxDeployment.DeepCopy()
+	redisDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
```

Comment:
imo deepcopying the deployment is a little confusing. Because you're updating the same rancher object each time, but storing different objects for nginx and redis in the code. imo it would be better to update the same object each time instead of copying it.

Reply:
I agree